packages/ui-keyring/src/index.ts

```diff
@@ -25,49 +27,94 @@ import getAddresses from './address/all';
 import saveAddress from './address/meta';
 import saveRecent from './address/metaRecent';
 
-const state: State = {
-  accounts,
-  addresses,
-  keyring: testKeyring()
-};
-
-loadAll(state);
-
-// FIXME The quicker we get in https://github.com/polkadot-js/apps/issues/138
-// the better, this is now completely out of control
-export default ({
-  addAccountPair: (pair: KeyringPair, password: string): KeyringPair =>
-    addAccountPair(state, pair, password),
-  createAccount: (seed: Uint8Array, password?: string, meta?: KeyringPair$Meta): KeyringPair =>
-    createAccount(state, seed, password, meta),
-  forgetAccount: (address: string): void =>
-    forgetAccount(state, address),
-  forgetAddress: (address: string): void =>
-    forgetAddress(state, address),
-  isAvailable: (address: string | Uint8Array): boolean =>
-    isAvailable(state, address),
-  getAccounts: (): Array<KeyringAddress> =>
-    getAccounts(state),
-  getAddress: (address: string | Uint8Array): KeyringAddress =>
-    getAddress(state, address),
-  getAddresses: (): Array<KeyringAddress> =>
-    getAddresses(state),
-  getPair: (address: string | Uint8Array): KeyringPair =>
-    state.keyring.getPair(address),
-  getPairs: (): Array<KeyringPair> =>
-    state.keyring.getPairs().filter((pair) =>
+class Keyring implements KeyringInstance {
+  private state: State;
+
+  constructor () {
+    this.state = {
+      accounts,
+      addresses,
+      keyring: testKeyring()
+    };
+
+    this.loadAll();
+  }
+
+  addPair (json: KeyringPair$Json): void {
+    return addPair(this.state, json);
```

Comment:
Ok, so we are half-way there. Bring the actual implementation in here, so we don't have the functionality scaterred accross, i.e. actual implementation goes inside this method, the state is actually on the class.